File: novi-devops-eindopdracht.client/src/app/services/book.service.ts
/* eslint-disable @angular-eslint/prefer-inject */
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Book } from '../models/book';
import { BookStatus } from '../models/book-status.enum';
import { environment } from '../../environments/environment';

@Injectable({ providedIn: 'root' })

export class BookService {
  private api = `${environment.apiUrl}/api/books`;

  constructor(private http: HttpClient) { }

  getAll(): Observable<Book[]> {
    return this.http.get<Book[]>(this.api);
  }

  add(book: Book) {
    return this.http.post(this.api, book);
  }

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  updateStatus(id: number, status: BookStatus): Observable<any> {
    return this.http.put(`${this.api}/${id}/status?status=${status}`, {});
  }

  getById(id: number): Observable<Book> {
    return this.http.get<Book>(`${this.api}/${id}`);
  }

  getByStatus(status: BookStatus): Observable<Book[]> {
    return this.http.get<Book[]>(`${this.api}/status/${status}`);
  }
}

File: novi-devops-eindopdracht.Server/Services/BookService.cs
using novi_devops_eindopdracht.Server.Models;

namespace novi_devops_eindopdracht.Server.Services
{
    public class BookService
    {
        private readonly List<Book> _books = new();

        public List<Book> GetAll()
        {
            return _books;
        }

        public Book? GetById(int id)
        {
            return _books.FirstOrDefault(b => b.Id == id);
        }

        public List<Book> GetByStatus(BookStatus status)
        {
            var bookList = _books.Where(b => b.Status == status).ToList();
            return bookList;
        }

        public Book Add(Book book)
        {
            book.Id = _books.Count + 1;
            _books.Add(book);
            return book;
        }

        public void UpdateStatus(int id, BookStatus status)
        {
            var book = GetById(id);
            if (book != null)
                book.Status = status;
        }
    }
}
